scripts/protocgen.sh

```diff
@@ -10,130 +10,26 @@ set -eo pipefail
-  else
-      echo "Fail - Current directory is not 'nibiru'"
-      exit 1  # Error exit status
-  fi
-
-  # Get protoc executions
-  go get github.com/regen-network/cosmos-proto/protoc-gen-gocosmos@latest 
-
-  # Get gogo/protobuf. This isn't always necessary. Newer versions of the CSDK
-  #   keep a copy of dependencies in the proto/third_party directory.
-  # go get github.com/gogo/protobuf/proto 2>/dev/null
-  
-  # Get Cosmos-SDK package from github
-  go get github.com/cosmos/cosmos-sdk@$SDK_VERSION 2>/dev/null
-}
-
-echo_pwd() {
-  echo "current dir: $(pwd)"
-}
-
-skip_line() {
-  printf "\n--------------------------------------------------\n"
-}
-
-# code_gen (fn): Generates code in TypeScript based on the proto files from
-#   the Cosmos-SDK (CSDK) and NibiruChain/nibiru repositories based on the 
-#   dependencies in the go.mod of Nibiru.
-code_gen() {
-  echo "Grabbing cosmos-sdk proto file locations from disk"
-  echo_pwd
-  cd ../nibiru;
-  protoc_gen_gocosmos
-  local cosmos_sdk_dir=$(go list -f '{{ .Dir }}' -m github.com/cosmos/cosmos-sdk)
-  echo "Happy: Found cosmos_sdk_dir: $cosmos_sdk_dir"
-
-  skip_line
-  echo "Grabbing all of the Cosmos-SDK and third party protos"
-  echo_pwd
-  cd -;
-  echo_pwd
-
-  local proto_dirs=$(find $cosmos_sdk_dir/proto $cosmos_sdk_dir/third_party/proto proto -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | uniq | sort)
-  echo "Proto Directories: "
-  for dir in $proto_dirs; do \
-    echo $dir
-  done;
-
-  skip_line
-  create_pkg_out_dir
-  echo "Clearing PKG_OUT_DIR: $PKG_OUT_DIR"
-  rm -rf $PKG_OUT_DIR/*
-
-  skip_line
-  for dir in $proto_dirs; do \
-    prefix=$HOME/go/pkg/mod/github.com
-    prefix_removed_string=${dir/#$prefix}
-    echo "generating $prefix_removed_string -----------------------------------" 
-    protoc \
-      --plugin=./node_modules/.bin/protoc-gen-ts_proto \
-      -I "$cosmos_sdk_dir/third_party/proto" \
-      -I "$cosmos_sdk_dir/proto" \
-      -I proto \
-      --ts_proto_opt="esModuleInterop=true,forceLong=long,useOptionals=messages" \
-      --ts_proto_out=$PKG_OUT_DIR \
-      $(find "${dir}" -type f -name '*.proto')
-  done; \
-}
 
 # -----------------------------------------------------------------------------
 # MAIN - Beginning of Script Execution
 # -----------------------------------------------------------------------------
-
-main () {
-  check_for_nibiru_repo
-  copy_protos_from_nibiru_repo
-  code_gen
-
-  rm go.mod go.sum
-  rm -rf proto
-}
-
-main
+echo "Removing old src files"
+rm -rf $PKG_OUT_DIR/
+mkdir -p $PKG_OUT_DIR/
+
+echo "Copying protos from the 'nibiru' repo"
+rm -rf proto/
+mkdir -p proto/
+cp -r $NIBIRU_REPO/proto/ proto/
+
+for dir in $(find ./proto -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | uniq | sort); do \
+  for file in $(find "${dir}" -maxdepth 1 -name '*.proto'); do
+    echo "Generating ts proto code for $file"
+    buf generate --template proto/buf.gen.ts.yaml -o $PKG_OUT_DIR $file
```

Comment:
Could use `npx buf` instead of `buf` so you don't have to install their CLI to make it work